From 3cae7953ccd2dcd6f01a7a3df372b6a19b9a2957 Mon Sep 17 00:00:00 2001
From: Shimizu Hikaru <MuPendulum@users.noreply.github.com>
Date: Tue, 11 Oct 2022 00:00:00 -0000
Subject: [PATCH] libretro sameboy: add optimized rk3326 platform

---
 libretro/Makefile | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/libretro/Makefile b/libretro/Makefile
index 4fec130..6bf9d3f 100644
--- a/libretro/Makefile
+++ b/libretro/Makefile
@@ -225,6 +225,21 @@ else ifeq ($(platform), vita)
    CFLAGS += -Wl,-q -O3 -fno-short-enums -fno-optimize-sibling-calls
 	STATIC_LINKING = 1
 
+# RK3326 and RK3399 devices
+else ifeq ($(platform),$(filter $(platform),rk3326 rk3399))
+   TARGET := $(TARGET_NAME)_libretro.so
+   fpic := -fPIC
+   SHARED := -shared -Wl,--version-script=$(CORE_DIR)/libretro/link.T
+   ifeq ($(platform), rk3326)
+      CPUFLAGS := -mcpu=cortex-a35+crc+crypto --param l1-cache-size=32 --param l1-cache-line-size=64 --param l2-cache-size=256
+   else ifeq ($(platform), rk3399)
+      CPUFLAGS := -mcpu=cortex-a72.cortex-a53+crc+crypto
+   endif
+   LTOFLAGS := -flto=auto -ffat-lto-objects -fdevirtualize-at-ltrans
+   CFLAGS += $(CPUFLAGS) $(LTOFLAGS) -Ofast -fomit-frame-pointer -fno-semantic-interposition -fipa-pta -fweb -frename-registers
+   CXXFLAGS += $(CFLAGS)
+   LDFLAGS += -Wl,-O1,--sort-common,--as-needed $(LTOFLAGS)
+
 # Windows MSVC 2017 all architectures
 else ifneq (,$(findstring windows_msvc2017,$(platform)))
 
@@ -336,7 +351,7 @@ LDFLAGS += $(LIBM)
 ifeq ($(DEBUG), 1)
    CFLAGS += -O0 -g
 else
-   CFLAGS += -O2 -DNDEBUG
+   CFLAGS += -DNDEBUG
 endif
 
 include Makefile.common
-- 
2.38.0

