From f8b38663bb8e70e01da829e875448449dacb8c00 Mon Sep 17 00:00:00 2001
From: Shimizu Hikaru <MuPendulum@users.noreply.github.com>
Date: Sun, 31 Dec 2023 00:00:00 +0000
Subject: [PATCH] libretro nestopia: add optimized rk3326 platform

---
 libretro/Makefile | 20 ++++++++++++++++++--
 1 file changed, 18 insertions(+), 2 deletions(-)

diff --git a/libretro/Makefile b/libretro/Makefile
index 7cfd5b0..74cefa9 100644
--- a/libretro/Makefile
+++ b/libretro/Makefile
@@ -543,6 +543,21 @@ else ifeq ($(platform),emscripten)
 	TARGET := $(TARGET_NAME)_libretro_$(platform).bc
 	STATIC_LINKING = 1
 
+# RK3326 and RK3399 devices
+else ifeq ($(platform),$(filter $(platform),rk3326 rk3399))
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+	SHARED := -shared -Wl,-version-script=link.T
+	ifeq ($(platform), rk3326)
+		CPUFLAGS := -mcpu=cortex-a35+crc+crypto --param l1-cache-size=32 --param l1-cache-line-size=64 --param l2-cache-size=256
+	else ifeq ($(platform), rk3399)
+		CPUFLAGS := -mcpu=cortex-a72.cortex-a53+crc+crypto
+	endif
+	LTOFLAGS := -fno-lto
+	CFLAGS += $(CPUFLAGS) $(LTOFLAGS) -O3 -fomit-frame-pointer -fno-semantic-interposition -fipa-pta -fweb -frename-registers
+	CXXFLAGS += $(CFLAGS)
+	LDFLAGS += -Wl,-O1,--sort-common,--as-needed $(LTOFLAGS) -static-libstdc++
+
 # Windows MSVC 2017 all architectures
 else ifneq (,$(findstring windows_msvc2017,$(platform)))
 
@@ -734,8 +749,8 @@ ifneq (,$(findstring msvc,$(platform)))
 CFLAGS   += -MT
 CXXFLAGS += -MT
 endif
-CFLAGS   += -O2 -DNDEBUG
-CXXFLAGS += -O2 -DNDEBUG
+CFLAGS   += -DNDEBUG
+CXXFLAGS += -DNDEBUG
 endif
 
 CORE_DIR = ..
@@ -754,6 +769,7 @@ endif
 DEFINES := -D__LIBRETRO__ $(PLATFORM_DEFINES) $(GCC_FLAGS) $(GCC_WARNINGS) -DNST_NO_ZLIB $(INCFLAGS) $(INCFLAGS_PLATFORM)
 
 CFLAGS += $(fpic) $(DEFINES) $(C_VER)
+CXXFLAGS := $(filter-out $(C_VER),$(CXXFLAGS))
 CXXFLAGS += $(fpic) $(DEFINES)
 
 INCDIRS := -I$(CORE_DIR) -I$(CORE_DIR)/source
