From 92d75db182b063a4dc9f97c1c7d4a5c74a72d4ba Mon Sep 17 00:00:00 2001
From: Shimizu Hikaru <MuPendulum@users.noreply.github.com>
Date: Mon, 4 Oct 2022 00:00:00 -0000
Subject: [PATCH] libretro gambatte: add optimized rk3326 platform

---
 Makefile.libretro | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/Makefile.libretro b/Makefile.libretro
index ffaba21..c16a23c 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -405,6 +405,19 @@ else ifeq ($(platform), miyoo)
    CFLAGS += -fomit-frame-pointer -ffast-math -march=armv5te -mtune=arm926ej-s
    CXXFLAGS += $(CFLAGS)
 
+# RK3326 devices
+else ifeq ($(platform), rk3326)
+   CPUFLAGS := -mcpu=cortex-a35+crypto -O3 -fno-plt -fno-semantic-interposition -ftracer -fgcse-sm -fgcse-las -fipa-pta -DNDEBUG
+   LTOFLAGS := -flto=auto -ffat-lto-objects -fdevirtualize-at-ltrans
+   CACHESIZES := --param l1-cache-size=32 --param l1-cache-line-size=64 --param l2-cache-size=256
+   TARGET := $(TARGET_NAME)_libretro.so
+   fpic := -fPIC
+   SHARED := -shared -Wl,--version-script=$(version_script)
+   CFLAGS += $(CPUFLAGS) $(LTOFLAGS) $(CACHESIZES)
+   CXXFLAGS += $(CPUFLAGS) $(LTOFLAGS) $(CACHESIZES)
+   LDFLAGS += -Wl,-O1,--sort-common,--as-needed -flto=auto -static-libstdc++
+   HAVE_NETWORK = 1
+
 # Windows MSVC 2003 Xbox 1
 else ifeq ($(platform), xbox1_msvc2003)
 TARGET := $(TARGET_NAME)_libretro_xdk1.lib
@@ -647,8 +660,8 @@ ifneq (,$(findstring msvc,$(platform)))
    CFLAGS   += -MT
    CXXFLAGS += -MT
 endif
-   CFLAGS   += -O2 -DNDEBUG
-   CXXFLAGS += -O2 -DNDEBUG
+   CFLAGS   += -DNDEBUG
+   CXXFLAGS += -DNDEBUG
 endif
 
 ifeq (,$(findstring msvc,$(platform)))
-- 
2.37.3

