From 6cd7835d27a16f4e86e4df94e30735c509897340 Mon Sep 17 00:00:00 2001
From: Shimizu Hikaru <MuPendulum@users.noreply.github.com>
Date: Sun, 31 Dec 2023 00:00:00 +0000
Subject: [PATCH] libretro gambatte: add optimized rk3326 platform

---
 Makefile.libretro | 20 ++++++++++++++++++--
 1 file changed, 18 insertions(+), 2 deletions(-)

diff --git a/Makefile.libretro b/Makefile.libretro
index 98ee91b..8f8ce3f 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -407,6 +407,22 @@ else ifeq ($(platform), miyoo)
    CFLAGS += -fomit-frame-pointer -ffast-math -march=armv5te -mtune=arm926ej-s
    CXXFLAGS += $(CFLAGS)
 
+# RK3326 and RK3399 devices
+else ifeq ($(platform),$(filter $(platform),rk3326 rk3399))
+   TARGET := $(TARGET_NAME)_libretro.so
+   fpic := -fPIC
+   SHARED := -shared -Wl,-version-script=$(version_script)
+   HAVE_NETWORK=1
+   ifeq ($(platform), rk3326)
+      CPUFLAGS := -mcpu=cortex-a35+crc+crypto --param l1-cache-size=32 --param l1-cache-line-size=64 --param l2-cache-size=256
+   else ifeq ($(platform), rk3399)
+      CPUFLAGS := -mcpu=cortex-a72.cortex-a53+crc+crypto
+   endif
+   LTOFLAGS := -flto=auto -ffat-lto-objects -fdevirtualize-at-ltrans
+   CFLAGS += $(CPUFLAGS) $(LTOFLAGS) -O3 -fomit-frame-pointer -fno-semantic-interposition -fipa-pta -fweb -frename-registers
+   CXXFLAGS += $(CFLAGS)
+   LDFLAGS += -Wl,-O1,--sort-common,--as-needed $(LTOFLAGS) -static-libstdc++
+
 # Windows MSVC 2003 Xbox 1
 else ifeq ($(platform), xbox1_msvc2003)
 TARGET := $(TARGET_NAME)_libretro_xdk1.lib
@@ -649,8 +665,8 @@ ifneq (,$(findstring msvc,$(platform)))
    CFLAGS   += -MT
    CXXFLAGS += -MT
 endif
-   CFLAGS   += -O2 -DNDEBUG
-   CXXFLAGS += -O2 -DNDEBUG
+   CFLAGS   += -DNDEBUG
+   CXXFLAGS += -DNDEBUG
 endif
 
 ifeq (,$(findstring msvc,$(platform)))
