From 41ff7918f2c8f6f24760498aa6d1e2a5eac4f426 Mon Sep 17 00:00:00 2001
From: Shimizu Hikaru <MuPendulum@users.noreply.github.com>
Date: Sun, 31 Dec 2023 00:00:00 +0000
Subject: [PATCH] libretro fbneo: add optimized rk3326 platform

---
 src/burner/libretro/Makefile | 23 +++++++++++++++++++++--
 1 file changed, 21 insertions(+), 2 deletions(-)

diff --git a/src/burner/libretro/Makefile b/src/burner/libretro/Makefile
index 7d7002b..2a50ae2 100644
--- a/src/burner/libretro/Makefile
+++ b/src/burner/libretro/Makefile
@@ -402,6 +402,24 @@ else ifeq ($(platform), emscripten)
 	EXTERNAL_ZLIB = 1
 	STATIC_LINKING = 1
 
+# RK3326 and RK3399 devices
+else ifeq ($(platform),$(filter $(platform),rk3326 rk3399))
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+	SHARED := -shared -Wl,--version-script=$(VERSION_SCRIPT)
+	ENDIANNESS_DEFINES := -DLSB_FIRST
+	LDFLAGS += -lpthread
+	EXTERNAL_ZLIB = 1
+	ifeq ($(platform), rk3326)
+		CPUFLAGS := -mcpu=cortex-a35+crc+crypto --param l1-cache-size=32 --param l1-cache-line-size=64 --param l2-cache-size=256
+	else ifeq ($(platform), rk3399)
+		CPUFLAGS := -mcpu=cortex-a72.cortex-a53+crc+crypto
+	endif
+	LTOFLAGS := -flto=auto -ffat-lto-objects -fdevirtualize-at-ltrans
+	CFLAGS += $(CPUFLAGS) $(LTOFLAGS) -Ofast -fomit-frame-pointer -fno-semantic-interposition -fipa-pta -fweb -frename-registers
+	CXXFLAGS += $(CFLAGS)
+	LDFLAGS += -Wl,-O1,--sort-common,--as-needed $(LTOFLAGS) -static-libstdc++
+
 # Windows MSVC 2017 all architectures
 else ifneq (,$(findstring windows_msvc2017,$(platform)))
 
@@ -582,8 +600,8 @@ ifeq ($(DEBUG), 1)
 else
 	ifeq (,$(findstring msvc,$(platform)))
 		# sadly, we can't use -Os for memory constrained systems here, the performance impact is too big
-		CFLAGS += -O3 -DNDEBUG -fomit-frame-pointer
-		CXXFLAGS += -O3 -DNDEBUG -fomit-frame-pointer
+		CFLAGS += -DNDEBUG -fomit-frame-pointer
+		CXXFLAGS += -DNDEBUG -fomit-frame-pointer
 	else
 		CFLAGS += -O2 -DNDEBUG
 		CXXFLAGS += -O2 -DNDEBUG
@@ -592,6 +610,7 @@ endif
 
 CFLAGS += $(fpic) $(FBNEO_DEFINES)
 CXXFLAGS += $(fpic) $(FBNEO_DEFINES)
+CXXFLAGS := $(filter-out -ansi -std=c90 -std=c99 -std=c11 -std=c17 -std=gnu90 -std=gnu99 -std=gnu11 -std=gnu17,$(CXXFLAGS))
 LDFLAGS += $(fpic)
 
 ifeq (,$(findstring msvc,$(platform)))
