From 06d01340c56b0089cedb889ed50f801e2d657e2d Mon Sep 17 00:00:00 2001
From: Shimizu Hikaru <MuPendulum@users.noreply.github.com>
Date: Tue, 4 Oct 2022 00:00:00 -0000
Subject: [PATCH] libretro snes9x: add optimized rk3326 platform

---
 libretro/Makefile        | 17 +++++++++++++++--
 libretro/Makefile.common |  5 +++++
 2 files changed, 20 insertions(+), 2 deletions(-)

diff --git a/libretro/Makefile b/libretro/Makefile
index 85fe350..f25bb09 100644
--- a/libretro/Makefile
+++ b/libretro/Makefile
@@ -361,6 +361,19 @@ else ifeq ($(platform), emscripten)
    STATIC_LINKING=1
    STATIC_LINKING_LINK=1
 
+# RK3326 devices
+else ifeq ($(platform), rk3326)
+   CPUFLAGS := -mcpu=cortex-a35+crc+crypto -Ofast -fomit-frame-pointer -fno-semantic-interposition -fipa-pta -ftree-vectorize -fweb -frename-registers -ftracer -DNDEBUG
+   LTOFLAGS := -flto=auto -ffat-lto-objects -fdevirtualize-at-ltrans
+   CACHESIZES := --param l1-cache-size=32 --param l1-cache-line-size=64 --param l2-cache-size=256
+   TARGET := $(TARGET_NAME)_libretro.so
+   fpic := -fPIC
+   SHARED := -shared -Wl,--version-script=link.T
+   CFLAGS += $(CPUFLAGS) $(LTOFLAGS) $(CACHESIZES) -DUNZIP_SUPPORT
+   CXXFLAGS += $(CPUFLAGS) $(LTOFLAGS) $(CACHESIZES) -DUNZIP_SUPPORT
+   LDFLAGS += -Wl,-O1,--sort-common,--as-needed -flto=auto -static-libstdc++
+   LIBS += -lz
+
 # Windows MSVC 2003 Xbox 1
 else ifeq ($(platform), xbox1_msvc2003)
 CFLAGS += -D__WIN32__
@@ -625,8 +638,8 @@ ifneq (,$(findstring msvc,$(platform)))
    CFLAGS += -O2 -DNDEBUG
    CXXFLAGS += -O2 -DNDEBUG
 else
-   CFLAGS += -O3 -DNDEBUG
-   CXXFLAGS += -O3 -DNDEBUG
+   CFLAGS += -DNDEBUG
+   CXXFLAGS += -DNDEBUG
 endif
 
    ifneq (,$(findstring msvc,$(platform)))
diff --git a/libretro/Makefile.common b/libretro/Makefile.common
index 79ae19c..7191d50 100644
--- a/libretro/Makefile.common
+++ b/libretro/Makefile.common
@@ -73,4 +73,9 @@ ifneq (,$(findstring win,$(platform)))
    INCFLAGS += -I$(CORE_DIR)/unzip
    SOURCES_C += $(CORE_DIR)/unzip/unzip.c \
                 $(CORE_DIR)/unzip/ioapi.c
+endif
+ifneq (,$(findstring rk3326,$(platform)))
+   INCFLAGS += -I$(CORE_DIR)/unzip
+   SOURCES_C += $(CORE_DIR)/unzip/unzip.c \
+                $(CORE_DIR)/unzip/ioapi.c
 endif
\ No newline at end of file
-- 
2.37.3

