From e60e206ebd4860f86b85b9a4c29ad3399ad380e8 Mon Sep 17 00:00:00 2001
From: Shimizu Hikaru <MuPendulum@users.noreply.github.com>
Date: Sun, 31 Dec 2023 00:00:00 +0000
Subject: [PATCH] libretro gpsp: add optimized rk3326 platform

---
 Makefile | 20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 5a3bab7..383612b 100644
--- a/Makefile
+++ b/Makefile
@@ -488,6 +488,24 @@ else ifeq ($(platform), miyoo)
 	HAVE_DYNAREC := 1
 	CPU_ARCH := arm
 
+# RK3326 and RK3399 devices
+else ifeq ($(platform),$(filter $(platform),rk3326 rk3399))
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+	SHARED := -shared -Wl,--version-script=link.T
+	HAVE_DYNAREC := 1
+	MMAP_JIT_CACHE = 1
+	CPU_ARCH := arm64
+	ifeq ($(platform), rk3326)
+		CPUFLAGS := -mcpu=cortex-a35+crc+crypto --param l1-cache-size=32 --param l1-cache-line-size=64 --param l2-cache-size=256
+	else ifeq ($(platform), rk3399)
+		CPUFLAGS := -mcpu=cortex-a72.cortex-a53+crc+crypto
+	endif
+	LTOFLAGS := -flto=auto -ffat-lto-objects -fdevirtualize-at-ltrans
+	CFLAGS += $(CPUFLAGS) $(LTOFLAGS) -Ofast -fomit-frame-pointer -fno-semantic-interposition -fipa-pta -fweb -frename-registers
+	CXXFLAGS += $(CFLAGS)
+	LDFLAGS += -Wl,-O1,--sort-common,--as-needed $(LTOFLAGS)
+
 # Windows
 else
 	TARGET := $(TARGET_NAME)_libretro.dll
@@ -512,7 +530,7 @@ endif
 ifeq ($(DEBUG), 1)
 	OPTIMIZE      := -O0 -g
 else
-	OPTIMIZE      := -O3 -DNDEBUG
+	OPTIMIZE      := -DNDEBUG
 endif
 
 DEFINES := -DHAVE_STRINGS_H -DHAVE_STDINT_H -DHAVE_INTTYPES_H -D__LIBRETRO__ -DINLINE=inline -Wall
