From b53986df505e406237bea6fbec2f1f85567ae1e5 Mon Sep 17 00:00:00 2001
From: Shimizu Hikaru <MuPendulum@users.noreply.github.com>
Date: Sun, 31 Dec 2023 00:00:00 +0000
Subject: [PATCH] libretro picodrive: add optimized rk3326 platform

---
 Makefile          |  2 +-
 Makefile.libretro | 22 ++++++++++++++++++++--
 2 files changed, 21 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 152e0181..1a0f313c 100644
--- a/Makefile
+++ b/Makefile
@@ -434,7 +434,7 @@ endif
 # want to remove this stuff for better performance if your compiler can handle it
 ifeq "$(DEBUG)" "0"
 ifeq (,$(findstring msvc,$(platform)))
-cpu/fame/famec.o: CFLAGS += -g0 -O2 -fno-expensive-optimizations
+#cpu/fame/famec.o: CFLAGS += -g0 -O2 -fno-expensive-optimizations
 else
 cpu/fame/famec.o: CFLAGS += -Od
 endif
diff --git a/Makefile.libretro b/Makefile.libretro
index 9e2511d0..e3965960 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -495,6 +495,24 @@ else ifeq ($(platform), miyoo)
 	HAVE_ARMv6 = 0
 	LOW_MEMORY = 1
 
+# RK3326 and RK3399 devices
+else ifeq ($(platform),$(filter $(platform),rk3326 rk3399))
+	TARGET := $(TARGET_NAME)_libretro.so
+	ARCH = aarch64
+	fpic := -fPIC
+	SHARED := -shared
+	PLATFORM_ZLIB = 0
+	CFLAGS += -DFAMEC_NO_GOTOS
+	ifeq ($(platform), rk3326)
+		CPUFLAGS := -mcpu=cortex-a35+crc+crypto --param l1-cache-size=32 --param l1-cache-line-size=64 --param l2-cache-size=256
+	else ifeq ($(platform), rk3399)
+		CPUFLAGS := -mcpu=cortex-a72.cortex-a53+crc+crypto
+	endif
+	LTOFLAGS := -flto=auto -ffat-lto-objects -fdevirtualize-at-ltrans
+	CFLAGS += $(CPUFLAGS) $(LTOFLAGS) -Ofast -fomit-frame-pointer -fno-semantic-interposition -fipa-pta -fweb -frename-registers
+	CXXFLAGS += $(CFLAGS)
+	LDFLAGS += -Wl,-O1,--sort-common,--as-needed $(LTOFLAGS)
+
 # Windows MSVC 2017 all architectures
 else ifneq (,$(findstring windows_msvc2017,$(platform)))
 	NO_GCC := 1
@@ -687,8 +705,8 @@ else
 		CFLAGS += -O2 -DNDEBUG
 		CXXFLAGS += -O2 -DNDEBUG
 	else
-		CFLAGS += -O3 -DNDEBUG
-		CXXFLAGS += -O3 -DNDEBUG
+		CFLAGS += -DNDEBUG
+		CXXFLAGS += -DNDEBUG
 	endif
 endif
 
