From e36d5ab670ec1f42d9eff91ac9fe6c7f9270994b Mon Sep 17 00:00:00 2001
From: Shimizu Hikaru <MuPendulum@users.noreply.github.com>
Date: Sun, 31 Dec 2023 00:00:00 +0000
Subject: [PATCH] libretro genesis plus gx: add optimized rk3326 platform

---
 Makefile.libretro | 22 ++++++++++++++++++++--
 1 file changed, 20 insertions(+), 2 deletions(-)

diff --git a/Makefile.libretro b/Makefile.libretro
index 58db740..c8437e3 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -488,6 +488,24 @@ else ifeq ($(platform), miyoo)
    LOW_MEMORY = 1
    MAX_ROM_SIZE = 16777216
 
+# RK3326 and RK3399 devices
+else ifeq ($(platform),$(filter $(platform),rk3326 rk3399))
+   TARGET := $(TARGET_NAME)_libretro.so
+   fpic := -fPIC
+   SHARED := -shared -Wl,--version-script=$(CORE_DIR)/libretro/link.T
+   ENDIANNESS_DEFINES := -DLSB_FIRST -DBYTE_ORDER=LITTLE_ENDIAN -DALIGN_LONG
+   PLATFORM_DEFINES := -DHAVE_ZLIB
+   MAX_ROM_SIZE = 33554432
+   ifeq ($(platform), rk3326)
+      CPUFLAGS := -mcpu=cortex-a35+crc+crypto --param l1-cache-size=32 --param l1-cache-line-size=64 --param l2-cache-size=256
+   else ifeq ($(platform), rk3399)
+      CPUFLAGS := -mcpu=cortex-a72.cortex-a53+crc+crypto
+   endif
+   LTOFLAGS := -flto=auto -ffat-lto-objects -fdevirtualize-at-ltrans
+   CFLAGS += $(CPUFLAGS) $(LTOFLAGS) -O3 -fomit-frame-pointer -fno-semantic-interposition -fipa-pta -fweb -frename-registers
+   CXXFLAGS += $(CFLAGS)
+   LDFLAGS += -Wl,-O1,--sort-common,--as-needed $(LTOFLAGS)
+
 # Windows MSVC 2003 Xbox 1
 else ifeq ($(platform), xbox1_msvc2003)
 TARGET := $(TARGET_NAME)_libretro_xdk1.lib
@@ -776,8 +794,8 @@ ifeq ($(platform), emscripten)
    CFLAGS   += -O3 -DNDEBUG
    CXXFLAGS += -O3 -DNDEBUG
 else
-   CFLAGS   += -O2 -DNDEBUG
-   CXXFLAGS += -O2 -DNDEBUG
+   CFLAGS   += -DNDEBUG
+   CXXFLAGS += -DNDEBUG
 endif
 endif
 
