From f5e08cb7609a9003b045bc0d625c7b65a0707822 Mon Sep 17 00:00:00 2001
From: Shimizu Hikaru <MuPendulum@users.noreply.github.com>
Date: Tue, 4 Oct 2022 00:00:00 -0000
Subject: [PATCH] libretro genesis plus gx: add optimized rk3326 platform

---
 Makefile.libretro | 19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/Makefile.libretro b/Makefile.libretro
index ed92c27..65fd83c 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -488,6 +488,21 @@ else ifeq ($(platform), miyoo)
    LOW_MEMORY = 1
    MAX_ROM_SIZE = 16777216
 
+# RK3326 devices
+else ifeq ($(platform), rk3326)
+   CPUFLAGS := -mcpu=cortex-a35+crc+crypto -O3 -fomit-frame-pointer -fno-semantic-interposition -fipa-pta -ftree-vectorize -fweb -frename-registers -ftracer -DNDEBUG
+   LTOFLAGS := -flto=auto -ffat-lto-objects -fdevirtualize-at-ltrans
+   CACHESIZES := --param l1-cache-size=32 --param l1-cache-line-size=64 --param l2-cache-size=256
+   TARGET := $(TARGET_NAME)_libretro.so
+   fpic := -fPIC
+   SHARED := -shared -Wl,--version-script=$(CORE_DIR)/libretro/link.T
+   CFLAGS += $(CPUFLAGS) $(LTOFLAGS) $(CACHESIZES)
+   CXXFLAGS += $(CPUFLAGS) $(LTOFLAGS) $(CACHESIZES)
+   LDFLAGS += -Wl,-O1,--sort-common,--as-needed -flto=auto
+   ENDIANNESS_DEFINES := -DLSB_FIRST -DBYTE_ORDER=LITTLE_ENDIAN -DALIGN_LONG
+   PLATFORM_DEFINES := -DHAVE_ZLIB
+   MAX_ROM_SIZE = 33554432
+
 # Windows MSVC 2003 Xbox 1
 else ifeq ($(platform), xbox1_msvc2003)
 TARGET := $(TARGET_NAME)_libretro_xdk1.lib
@@ -772,8 +787,8 @@ ifneq (,$(findstring msvc,$(platform)))
    CFLAGS   += -MT
    CXXFLAGS += -MT
 endif
-   CFLAGS   += -O2 -DNDEBUG
-   CXXFLAGS += -O2 -DNDEBUG
+   CFLAGS   += -DNDEBUG
+   CXXFLAGS += -DNDEBUG
 endif
 
 ifeq ($(SHARED_LIBVORBIS),)
-- 
2.38.0

