From 839a341e7399f0e37ca77fb5470ed41cf69f18e0 Mon Sep 17 00:00:00 2001
From: Shimizu Hikaru <MuPendulum@users.noreply.github.com>
Date: Sun, 31 Dec 2023 00:00:00 +0000
Subject: [PATCH] libretro pcsx rearmed: add optimized rk3326 platform

---
 Makefile          |  4 ++--
 Makefile.libretro | 19 +++++++++++++++++++
 2 files changed, 21 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 9b5d245b..f415732f 100644
--- a/Makefile
+++ b/Makefile
@@ -9,7 +9,7 @@ else
 ifeq ($(platform), $(filter $(platform), vita ctr))
 CFLAGS += -O3 -DNDEBUG
 else
-CFLAGS += -O2 -DNDEBUG
+CFLAGS += -DNDEBUG
 endif
 endif
 ifeq ($(DEBUG_ASAN), 1)
@@ -246,7 +246,7 @@ ifeq "$(THREAD_RENDERING)" "1"
 CFLAGS += -DTHREAD_RENDERING
 OBJS += plugins/gpulib/gpulib_thread_if.o
 endif
-plugins/gpu_unai/gpulib_if.o: CFLAGS += -DREARMED -O3 
+plugins/gpu_unai/gpulib_if.o: CFLAGS += -DREARMED
 CC_LINK = $(CXX)
 endif
 
diff --git a/Makefile.libretro b/Makefile.libretro
index d3a3530c..7304b6e5 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -568,6 +568,25 @@ else ifeq ($(platform), emscripten)
    STATIC_LINKING = 1
    HAVE_PHYSICAL_CDROM = 0
 
+# RK3326 and RK3399 devices
+else ifeq ($(platform),$(filter $(platform),rk3326 rk3399))
+	TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+	ARCH := arm64
+	BUILTIN_GPU = neon
+	HAVE_NEON = 1
+	DYNAREC = ari64
+	HAVE_PHYSICAL_CDROM = 0
+	ifeq ($(platform), rk3326)
+		CPUFLAGS := -mcpu=cortex-a35+crc+crypto --param l1-cache-size=32 --param l1-cache-line-size=64 --param l2-cache-size=256
+	else ifeq ($(platform), rk3399)
+		CPUFLAGS := -mcpu=cortex-a72.cortex-a53+crc+crypto
+	endif
+	LTOFLAGS := -flto=auto -ffat-lto-objects -fdevirtualize-at-ltrans
+	CFLAGS += $(CPUFLAGS) $(LTOFLAGS) -Ofast -fomit-frame-pointer -fno-semantic-interposition -fipa-pta -fweb -frename-registers
+	CXXFLAGS += $(CFLAGS)
+	LDFLAGS += -Wl,-O1,--sort-common,--as-needed $(LTOFLAGS)
+
 # Windows
 else
 	TARGET := $(TARGET_NAME)_libretro.dll
