From 946ffc87443ac527a9cf8d683221a92ad323106c Mon Sep 17 00:00:00 2001
From: Shimizu Hikaru <MuPendulum@users.noreply.github.com>
Date: Tue, 4 Oct 2022 00:00:00 -0000
Subject: [PATCH] libretro beetle ngp: add optimized rk3326 platform

---
 Makefile | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index a423a92..fb6a8af 100644
--- a/Makefile
+++ b/Makefile
@@ -349,6 +349,18 @@ else ifeq ($(platform), gcw0)
    SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
    FLAGS += -ffast-math -march=mips32 -mtune=mips32r2 -mhard-float
 
+# RK3326 devices
+else ifeq ($(platform), rk3326)
+   CPUFLAGS := -mcpu=cortex-a35+crc+crypto -O3 -fomit-frame-pointer -fno-semantic-interposition -fipa-pta -ftree-vectorize -fweb -frename-registers -ftracer -fno-plt -DNDEBUG
+   LTOFLAGS := -flto=auto -ffat-lto-objects -fdevirtualize-at-ltrans
+   CACHESIZES := --param l1-cache-size=32 --param l1-cache-line-size=64 --param l2-cache-size=256
+   TARGET := $(TARGET_NAME)_libretro.so
+   fpic := -fPIC
+   SHARED := -shared -Wl,--version-script=link.T
+   CFLAGS += $(CPUFLAGS) $(LTOFLAGS) $(CACHESIZES)
+   CXXFLAGS += $(CPUFLAGS) $(LTOFLAGS) $(CACHESIZES)
+   LDFLAGS += -Wl,-O1,--sort-common,--as-needed -flto=auto -static-libstdc++
+
 # Windows MSVC 2017 all architectures
 else ifneq (,$(findstring windows_msvc2017,$(platform)))
 
@@ -611,8 +623,8 @@ ifneq (,$(findstring msvc,$(platform)))
    CFLAGS   += -MT
    CXXFLAGS += -MT
 endif
-   CFLAGS   += -O2 -DNDEBUG
-   CXXFLAGS += -O2 -DNDEBUG
+   CFLAGS   += -DNDEBUG
+   CXXFLAGS += -DNDEBUG
 endif
 
 
-- 
2.38.0

