From 506b9c69be0c50996ccd51482e607c5cf2374edd Mon Sep 17 00:00:00 2001
From: Vicki Pfau <vi@endrift.com>
Date: Sat, 15 Oct 2022 23:01:36 -0700
Subject: [PATCH] GBA Video: Optimize mode 0 16-color tiles a bit more

---
 src/gba/renderers/software-mode0.c | 34 ++++++++++++------------------
 1 file changed, 14 insertions(+), 20 deletions(-)

diff --git a/src/gba/renderers/software-mode0.c b/src/gba/renderers/software-mode0.c
index 209032388..1736929c8 100644
--- a/src/gba/renderers/software-mode0.c
+++ b/src/gba/renderers/software-mode0.c
@@ -160,27 +160,21 @@
 			pixel += 8; \
 			continue; \
 		} \
-		LOAD_32(tileData, charBase, vram); \
+		if (!GBA_TEXT_MAP_HFLIP(mapData)) { \
+			LOAD_32(tileData, charBase, vram); \
+		} else { \
+			LOAD_32BE(tileData, charBase, vram); \
+			tileData = ((tileData & 0xF0F0F0F0) >> 4) | ((tileData & 0x0F0F0F0F) << 4); \
+		} \
 		if (tileData) { \
-			if (!GBA_TEXT_MAP_HFLIP(mapData)) { \
-				BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 0); \
-				BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 1); \
-				BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 2); \
-				BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 3); \
-				BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 4); \
-				BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 5); \
-				BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 6); \
-				BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 7); \
-			} else { \
-				BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 7); \
-				BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 6); \
-				BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 5); \
-				BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 4); \
-				BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 3); \
-				BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 2); \
-				BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 1); \
-				BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 0); \
-			} \
+			BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 0); \
+			BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 1); \
+			BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 2); \
+			BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 3); \
+			BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 4); \
+			BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 5); \
+			BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 6); \
+			BACKGROUND_DRAW_PIXEL_16(BLEND, OBJWIN, 7); \
 		} \
 		pixel += 8; \
 	}
